### PR TITLE
Merge feature/react-on-draxo-feedback into developpe

### PR DESCRIPTION
**Pull request by Francois389**
#177 

# Result 

Here we can see tree different possibility :

1. A review that was hidden by the teacher
2. A review with the utility grade `Tout a fais d'accord` has been selected
3. A review for which we don't have the report button because the evaluation has no comment. 

image

There is the teacher side.
We can notice the button for hidding the evaluation

image

There is the report modal.
When the `Autre` checkbox is checked, we have to specify a comment

image

If the student have reported an evaluation, he can't see it anymore.
If the evaluation is pointless, like for exemple the grader does'nt have an opinion or does'nt know, the student can't react to it.
image

### CODE REVIEW

**Review by FranckSilvestre — CHANGES_REQUESTED** [review]
- there is no feedback after a student has submitted a report or a utility grade on a peer review (the given grade is no displayed after submission)
- the student loose the focus when he submits a report or a utility grade.
- the teacher loose the focus when hiding a peer grading

File: src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/response/ResponseService.kt [hunk 1]
```diff
@@ -381,4 +382,26 @@ class ResponseService(
...
+
+
+    /**
+     * Return true if the user can moderate the feedback of the response
+     * An user can moderate the feedback if he is the owner of the sequence
+     * @param user the user who want to moderate the feedback
+     * @param response the response to moderate
+     */
+    fun canModerate(user: User, response: Response): Boolean {
+        return response.learner == user
```
[the remark this answers is not in the document]
Comment (FranckSilvestre) [reply src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/response/ResponseService.kt:405]: TODO: modify the doc and rename the method: canModerateFeedbackOfResponse

[the remark this answers is not in the document]
Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/response/ResponseService.kt:405]: the commit [e7712f9](https://github.com/elaastic/elaastic-questions-server/pull/235/commits/e7712f9b002a4a00f3b2294f2b35a1df0eba5e13) solved the issue...

File: src/main/resources/templates/player/assignment/sequence/phase/evaluation/method/draxo/_draxo-show-list.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
     <!-- Link to the css file for the custom DRAXO grid -->
     <link rel="stylesheet" type="text/css" th:href="@{'/css/draxo-show.css'}"/>
     <th:block th:each="model : ${evaluationModelList}">
-        <div th:replace="player/assignment/sequence/phase/evaluation/method/draxo/_draxo-show::draxoShow(${model},${hideName})"></div>
+        <div th:replace="player/assignment/sequence/phase/evaluation/method/draxo/_draxo-show::draxoShow(${model},${hideName}, ${canModeratePeerGrading}, ${canHidePeerGrading})"></div>
```
[the remark this answers is not in the document]
Comment (FranckSilvestre) [reply src/main/resources/templates/player/assignment/sequence/phase/evaluation/method/draxo/_draxo-show-list.html]: Processed after peer review

**Review by FranckSilvestre — APPROVED** [review]

**Pull request merged**